tests/test_network_share_lifecycle.py

```
@@ -0,0 +1,94 @@
+
+
+def test_share_listener_lifecycle():
+    # Ensure a clean starting state regardless of test ordering.
+    if ns.get_state().enabled:
+        asyncio.run(ns.disable(__import__("fastapi").FastAPI()))
+
+    try:
+        # Probe whether binding 0.0.0.0 is permitted in this sandbox.
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as probe:
+            try:
+                probe.bind(("0.0.0.0", 0))
```

Ruff (0.15.14)

[error] Possible binding to all interfaces

(S104)